sql/server/src/main/java/io/stargate/api/sql/plan/rel/CassandraFullScan.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.api.sql.plan.rel;
+
+import io.stargate.api.sql.plan.CalciteUtils;
+import io.stargate.api.sql.plan.exec.RuntimeContext;
+import io.stargate.api.sql.plan.exec.StatementExecutor;
+import io.stargate.api.sql.schema.StargateTable;
+import java.util.Collections;
+import org.apache.calcite.adapter.enumerable.*;
+import org.apache.calcite.linq4j.tree.BlockBuilder;
+import org.apache.calcite.linq4j.tree.Expression;
+import org.apache.calcite.linq4j.tree.Expressions;
+import org.apache.calcite.plan.*;
+import org.apache.calcite.rel.core.TableScan;
+import org.apache.calcite.rel.metadata.RelMetadataQuery;
+import org.apache.calcite.schema.Table;
+
+/** Relational operation representing a full Cassandra table scan. */
+public class CassandraFullScan extends TableScan implements EnumerableRel {
```

Comment:
I'd avoid the `Cassandra` prefix (here and other similar places), since storage is supposedly somewhat abstracted (I mean, storage pretty much has to be CQL-based, but it doesn't have to be "Cassandra"). Imo, `FullScan` is probably good enough, though if we're worried this might be confused by a "SQL full scan", maybe `StorageFullScan`.

Reply:
Yeah, "Cassandra" does feel awkward in this context. At the same time we certainly rely of C* storage model (PK/CC, etc.) in these classes. I'm fine with `StorageFullScan`.

Reply:
>  we certainly rely on C* storage model (PK/CC, etc.) in these classes

Right, but that's what Stargate does. I mean, `Persistence` is the core of Stargate, so something unqualified like `FullScan` would imo naturally imply in the reader mind "a full scan related to the persistence storage". But I'm happy with adding `Storage`.

Reply:
Another aspect of this is that these classes show up in `EXPLAIN PLAN`. We can probably keep `FullScan` as the class name, but it may be too ambiguous for the plan explanation... I'll see if we have a way to override what name is used in `EXPLAIN PLAN`.

Reply:
A did some renaming in these and related classes. Please have another look.

As for the "Stargate" prefix, the only remaining instance is `StargateMeta`... I really could not find a better fit. `MetaImpl` is already in its inheritance chain, plus it's a bit terse, and I wanted to emphasize that this impl. is a "Stargate" way of integrating with Avatica.

"Cassandra" prefixes are gone. That actually makes `EXPLAIN PLAN` output more readable I think (we already have physical table names there, so the association with "storage" is pretty clear.. see `QueryPlannerTest` for examples, if interested).